prediction_market_agent/agents/safe_guard_agent/app.py

```diff
@@ -47,31 +53,89 @@
     "Your safe address (or keep the default one provided)",
     value="gno:0x1275EE969cd187Fd7B9c6Aa8339d41985eFd1EF1",
 )
-do_execute = st.checkbox("Execute transaction if validated", value=False)
-do_reject = st.checkbox("Reject transaction if not validated", value=False)
-do_message = st.checkbox("Send a message about the outcome", value=False)
 
 if not safe_address:
     st.stop()
 
 # Get rid of potential token id
 *_, safe_address = safe_address.split(":")
 safe_address_checksum = Web3.to_checksum_address(safe_address)
+is_owner = check_if_owner(safe_address_checksum, keys.bet_from_address)
+
+do_execute = st.checkbox(
+    "Execute transaction if validated (possible only for Safe's owners)",
+    value=False,
+    disabled=not is_owner,
+)
+do_reject = st.checkbox(
+    "Reject transaction if not validated (possible only for Safe's owners)",
+    value=False,
+    disabled=not is_owner,
+)
```

Comment:
Do I understand correctly this is only displayed if agent is a signer? 
If e.g. we have the use case where a given proposed tx should be analyzed, then no reason to display this.

Reply:
If he isn't the owner, the checkbox is disabled but visible. A way to let people know that if they do make him a signer, they can try this out too.